### ISSUE TITLE
Proposal: Flow for booking a volunteer slot

### ISSUE THREAD

**Issue by road2nowhere** [where2help/creative#1]
WEB & APP
Home page of W2H should always be the list of current needs, no matter if people are logged in or not. If not logged in, an error/warning/info with link to login/registration can be presented when they try to book a slot.
![w2h_flow_booking](https://cloud.githubusercontent.com/assets/15609804/10913493/74906c18-824e-11e5-8602-e77fcf0d8e5f.png)

**Comment by markusrubey** [comment]
Totally agree on the flow, already started to implement it this way. :thumbsup: 

The second state won't be on mobile, to keep things simple, right? Instead i would like to show a mark if the user signed up for the need. Like so:
![device-2015-11-02-123744](https://cloud.githubusercontent.com/assets/2398384/10913876/1f7d1148-8250-11e5-9309-df50491e3daf.png)

**Comment by road2nowhere** [comment]
With "second state" you mean the detail view? 
We need it also on mobile since we have the description, map, link to routeplaner (?), and link to calendar entry there. It also gives some reassurance to the user.... I don't think it makes things too complicated, because it's a pretty standard behaviour, direct booking from a list a more unusual. We might need to change the button label though..

I like the checks :)
Is it realistic for the MVP to have also a place where I can see all my booked events (could be a fiter, but better accessible via menu)?

**Comment by markusrubey** [comment]
I mean having the "Helfen" button in the list on mobile. Of course we keep the details page! ;)

Already implemented on mobile. If you press on the upper right account icon you would get to a "profile-screen" where all booked items are listed. Sorry that the app keeps crashing on you, otherwise this would be much simpler. But if you take a look at the "old" flow you could get a sense of the current implementation: https://cloud.githubusercontent.com/assets/2398384/10636274/fe62312e-77fe-11e5-8533-7b7e90007e6c.jpg

**Comment by road2nowhere** [comment]
Ah, ok :))))

**Comment by klappradla** [comment]
> Is it realistic for the MVP to have also a place where I can see all my booked events

Signed in as a volunteer u only have one other menu item besides home: _Meine Termine_ - that's where u find all needs u volunteer and volunteered for. U can filter the view by time and place.

For the list of needs, we already had a discussion about the button and anti the check icon [here](https://github.com/where2help/where2help/issues/8). I personally think the _straight-sign-up-for-a-volunteering_ button sense in the web interface:
The web ui has more space available: why bother the user with an additional click and especially an additional navigation step back afterwards for signing in to a slog at e.g. Wien Westbahnhof? Anyone knows where it is and we'll send a reminder email anyways. The straight-sign-up-button lowers the barrier and facilitates browsing the feed and signing up to multiple slots. The detail view is not available yet, but will of course exist, too.

If getting rid of the straight-sign-in-button: what should we display there instead? The ui has much more space than the mobile view. The exact same view as mobile but just wieder may look ridiculous. We should show more information on the web ui than on the mobile.

**Comment by markusrubey** [comment]
I was speaking about the android app only. On the web i wouldn't change a thing, cause there is plenty of room as you said and it's pretty straight forward as of now. The "Absagen" state could be more differentiated then the "Helfen" state of the button in terms of color, but that's it. If the user needs more infos (desc, map, ...) he/she goes to detail and can check in there as well. What are you sceptical of @loncie ?

**Comment by road2nowhere** [comment]
Ok, interesting points. I think we can try if we find a solution that combines the benefits of both. I will work on the list tomorrow and come up with some proposals.

**Comment by markusrubey** [comment]
Maybe making the entire list item on the web clickable and having a highlight on that could make it more clear to the user that there is in fact a details-page?

**Comment by road2nowhere** [comment]
Yes, it's one possibility. I think we have various options to better differentiate how to get to the details and how to book, design wise and also with the copy (eg. something simple like changing "helfen" to "jetzt helfen" oder "ich helfe"). I agree with you that the space should be used efficiently on the web. 

Question: The map (open street map or Google) will be available also on the Web, right?

**Comment by road2nowhere** [comment]
@klappradla I think the checks in @markusrubey mobile screens are different from the ones you discussed previously, since they really represent where you have already signed up. They could be an option if we don't have a chance for the MVP to solve this more cleanly (i.e. clear the list of events I have already signed up for an make them available somewhere separately, like on the web in "Meine Termine")

**Comment by klappradla** [comment]
The map will be available on the web app (@ramonh is already working on it).

I also thought about clearing the items one signed up for first. But I then decided against, cause it removes necessary feedback for the user to know what's happening. It would have been necessary to show a flash, etc. telling "You signed in". So we would need to add content telling the user we removed other content.<br>Just increases movement on the screen, no real benefit in my eyes.
